.github/autobuild/windows.ps1

```diff
@@ -0,0 +1,169 @@
+    if (!$?)
+    {
+        echo "WARNING: Qt installation via first aqt run failed, re-starting with different base URL."
+        aqt install-qt -b https://mirrors.ocf.berkeley.edu/qt/ @Args
+        if (!$?)
+        {
+            throw "Qt installation with args @Args failed with exit code $LastExitCode"
+        }
+    }
+}
+
+Function Ensure-Qt
+{
+    if (Test-Path -Path $QtDir)
+    {
+        echo "Using Qt installation from previous run (actions/cache)"
+        return
+    }
+
+    echo "Install Qt..."
+    # Install Qt
+    pip install "aqtinstall==$AqtinstallVersion"
+    if (!$?)
+    {
+        throw "pip install aqtinstall failed with exit code $LastExitCode"
+    }
+
+    echo "Get Qt 64 bit..."
+    Install-Qt "${Qt64Version}" "${Msvc64Version}"
+
+    echo "Get Qt 32 bit..."
+    Install-Qt "${Qt32Version}" "${Msvc32Version}"
+}
+
+Function Ensure-jom
+{
+    choco install --no-progress -y jom --version "${JomVersion}"
+}
+
+Function Ensure-JACK
+{
+    if ($BuildOption -ne "jackonwindows")
+    {
+        return
+    }
+
+    echo "Install JACK2 64-bit..."
+    # Install JACK2 64-bit
+    choco install --no-progress -y jack --version "${JackVersion}"
+    if (!$?)
+    {
+        throw "64bit jack installation failed with exit code $LastExitCode"
+    }
+
+    echo "Install JACK2 32-bit..."
+    # Install JACK2 32-bit (need to force choco install as it detects 64 bits as installed)
+    choco install --no-progress -y -f --forcex86 jack --version "${JackVersion}"
+    if (!$?)
+    {
+        throw "32bit jack installation failed with exit code $LastExitCode"
+    }
+}
+
+Function Build
+{
+    echo "Build installer..."
+    $ExtraArgs = @()
+    if ($BuildOption -ne "")
+    {
+        $ExtraArgs += ("-BuildOption", $BuildOption)
+    }
+    powershell ".\windows\deploy_windows.ps1" "C:\Qt\5.15.2" "C:\Qt\5.15.2" @ExtraArgs
+    if (!$?)
+    {
+        throw "deploy_windows.ps1 failed with exit code $LastExitCode"
+    }
+}
+
+Function Pass-Artifact-to-Job
+{
+    # Add $BuildOption as artifact file name suffix. Shorten "jackonwindows" to just "jack":
+    $ArtifactSuffix = switch -Regex ($BuildOption)
+    {
+        "jackonwindows" { "_jack"; break }
+        "^\S+$"         { "_${BuildOption}"; break }
+        default         { "" }
+    }
+
+    $artifact_deploy_filename = "jamulus_${JamulusVersion}_win${ArtifactSuffix}.exe"
+
+    echo "Copying artifact to ${artifact_deploy_filename}"
+    cp ".\deploy\Jamulus*installer-win.exe" ".\deploy\${artifact_deploy_filename}"
+    if (!$?)
+    {
+            throw "cp failed with exit code $LastExitCode"
+    }
+    echo "Setting Github step output name=artifact_1::${artifact_deploy_filename}"
+    echo "::set-output name=artifact_1::${artifact_deploy_filename}"
+}
+
+switch ($Stage)
+{
+    "setup"
+    {
+        choco config set cacheLocation $ChocoCacheDir
+        Ensure-Qt
+        Ensure-jom
+        Ensure-JACK
+    }
+    "build"
+    {
+        Build
```

Comment:
This looks really strange to me (the Powershell syntax): Maybe add a short comment if we plan to continue going the 3 step build. But as I understood, you'd like to move away - so it doesn't matter now probably

Reply:
I think they'll stay at 2 (setup, build) or even 3 steps (as it is). The initial autobuild logic's reason for separate steps seems to have been the added value regarding runtime statistics. That might still be interesting, so we should probably keep it.

What would you want to have commented exactly? I'm pretty new to PowerShell, but this part sounds like an ordinary switch statement to me?
https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-switch?view=powershell-7.2

I've just realized that the official docs use the `switch ( foo )` spacing style (similar to what we do for C++), so I've just adapted this to be consistent as well.

Reply:
In my opinion it looks a bit like that:

switch ($switch)
  case 'switch': switch

I'd prefer something like

"build" {
  # Run deploy script to build Jamulus and installer
  Build
}

The comment is basically just there to make a clear cut between the case and the function.

Reply:
Ah, now I understand. I've just renamed the function -- does that make it more readable?

Reply:
Yes.